src/NLog/Internal/Reflection/PropertyHelper.cs

```diff
@@ -103,21 +103,31 @@ internal static void SetPropertyFromString(object obj, string propertyName, stri
             try
             {
                 Type propertyType = propInfo.PropertyType;
+                var unwrappedType = UnwrapTypedLayout(propertyType);
+                var isTypedLayout = unwrappedType != propertyType;
 
-                if (!TryNLogSpecificConversion(propertyType, value, configurationItemFactory, out var newValue))
+                if (!TryNLogSpecificConversion(unwrappedType, value, configurationItemFactory, out var newValue))
                 {
                     if (propInfo.IsDefined(_arrayParameterAttribute.GetType(), false))
                     {
                         throw new NotSupportedException($"Parameter {propertyName} of {objType.Name} is an array and cannot be assigned a scalar value.");
                     }
 
-                    propertyType = Nullable.GetUnderlyingType(propertyType) ?? propertyType;
+                    unwrappedType = Nullable.GetUnderlyingType(unwrappedType) ?? unwrappedType;
```

Comment:
Maybe move this logic into `UnwrapTypedLayout()`

Reply:
not possible, we need the prev value of unwrappedType above